crates/phactory/src/contracts/geolocation.rs

```diff
@@ -124,39 +136,50 @@ impl contracts::NativeContract for Geolocation {
         req: Request,
     ) -> Result<Response, Error> {
         match req {
-            Request::GetGeolocationInfo { account } => {
+            Request::GetGeocoding { account } => {
                 if origin != Some(&account) {
                     return Err(Error::NotAuthorized);
                 }
-                if let Some(data) = self.geolocation_info.get(&account) {
-                    let geolocation_info = data.clone();
-                    Ok(Response::GetGeolocationInfo { geolocation_info })
-                } else {
-                    Err(Error::NoRecord)
-                }
+                let geo_data = self.geo_data.get(&account).ok_or(Error::NoRecord)?;
+                let geocoding = geo_data.data.clone();
+                Ok(Response::GetGeocoding { geocoding })
             }
-            Request::GetAvailableCityName {} => {
-                let city_names: Vec<String> = self.city_distribution.keys().cloned().collect();
-                Ok(Response::GetAvailableCityName { city_names })
+            Request::GetAvailableRegionName {} => {
+                // let valid_region_map = self.region_map.clone()
```

Comment:
It's not possible anymore. Safe to remove

Reply:
I think this request is for querying what regions we have and thus is always possible. Consider a circumstance where there are 2 workers respectively from New York and Shang Hai in the network. Since there have other requests requiring region name as a input, If we don't querying what regions we have first, we have to only "guess" or “enumerate” all possible regions to see if there are any worker in that region. However, by utilizing this method we can know there are New York and Shang Hai where workers exist.

Reply:
Moreover, currently Geolite doesn't provide a full city list (https://support.maxmind.com/geoip-faq/specifications-and-implementation/which-cities-are-in-the-geoip2-and-geoip-legacy-city-databases/ ), so there is no guidance for workers on how to enumerate all possible region names. In fact, we can only derive a full city list by parsing the database in advance, which I think is neither clean nor safe.

Reply:
Oh, I meant the code in the comments can be removed. It was used to filter out empty regions.

Reply:
Ohh, got it. ;)